pkg/controller/deployment/unsupported_config_overrides.go

```diff
@@ -1,23 +1,63 @@
 package deployment
 
 import (
+	"fmt"
+	"strings"
+
 	appsv1 "k8s.io/api/apps/v1"
 
 	"github.com/openshift/cert-manager-operator/api/operator/v1alpha1"
 )
 
+const argKeyValSeparator = "="
+
 func UnsupportedConfigOverrides(deployment *appsv1.Deployment, unsupportedConfigOverrides *v1alpha1.UnsupportedConfigOverrides) *appsv1.Deployment {
 	if unsupportedConfigOverrides == nil {
 		return deployment
 	}
 	if len(unsupportedConfigOverrides.Webhook.Args) > 0 && deployment.Name == "cert-manager-webhook" {
-		deployment.Spec.Template.Spec.Containers[0].Args = unsupportedConfigOverrides.Webhook.Args
+		deployment.Spec.Template.Spec.Containers[0].Args = overrideConfigArgs(
+			unsupportedConfigOverrides.Webhook.Args,
+			deployment.Spec.Template.Spec.Containers[0].Args)
 	}
 	if len(unsupportedConfigOverrides.CAInjector.Args) > 0 && deployment.Name == "cert-manager-cainjector" {
-		deployment.Spec.Template.Spec.Containers[0].Args = unsupportedConfigOverrides.CAInjector.Args
+		deployment.Spec.Template.Spec.Containers[0].Args = overrideConfigArgs(
+			unsupportedConfigOverrides.CAInjector.Args,
+			deployment.Spec.Template.Spec.Containers[0].Args)
 	}
 	if len(unsupportedConfigOverrides.Controller.Args) > 0 && deployment.Name == "cert-manager" {
-		deployment.Spec.Template.Spec.Containers[0].Args = unsupportedConfigOverrides.Controller.Args
+		deployment.Spec.Template.Spec.Containers[0].Args = overrideConfigArgs(
+			unsupportedConfigOverrides.Controller.Args,
+			deployment.Spec.Template.Spec.Containers[0].Args)
 	}
 	return deployment
 }
+
+func overrideConfigArgs(overrideArgs []string, sourceArgs []string) (destArgs []string) {
+	destArgMap := map[string]string{}
+	updateArgMap(destArgMap, sourceArgs)
+	updateArgMap(destArgMap, overrideArgs)
+
+	destArgs = make([]string, len(destArgMap))
+	i := 0
+	for key, val := range destArgMap {
+		if len(val) > 0 {
+			destArgs[i] = fmt.Sprintf("%s%s%s", key, argKeyValSeparator, val)
+		} else {
+			destArgs[i] = key
+		}
+		i++
+	}
+	return destArgs
+}
+
+func updateArgMap(argMap map[string]string, args []string) {
+	for _, arg := range args {
+		splitted := strings.Split(arg, argKeyValSeparator)
```

Comment:
I say we skip on empty `arg` instead of adding empty entries into the map.

Reply:
Added a if block to guard around empty args like `if len(splitted) > 0 { ... }` which should be able to take care of that.

Reply:
Also, just to clarify in case you were referring to args containing only the key (`--key`) without any value as an empty arg, in that case updating the map with atleast the key would still be required otherwise we would end up missing such args defeating the purpose of the map.

Reply:
Hmm, what I meant by empty `arg` was if `args={"", "--foo", "--foo=bar"}`. Then again the `len()` would cover this case.

Reply:
Yeah, got it. I checked by putting `""` in the args list via `TestParseArgMap`, the len was still 1. Fixed it by updating the if statement to use: `if len(splitted) > 0 && arg != "" { ... }` instead, which helps achieve desired functionality now.